src/pkg/packager/composer/list.go

```diff
@@ -129,78 +136,69 @@ func NewImportChain(head types.ZarfComponent, index int, arch, flavor string) (*
+			return ic, fmt.Errorf("detected malformed import chain, cannot import remote components from remote components")
 		}
 		// ensure that remote components are not importing local components
 		if node.prev != nil && node.prev.Import.URL != "" && isLocal {
-			return nil, fmt.Errorf("detected malformed import chain, cannot import local components from remote components")
+			return ic, fmt.Errorf("detected malformed import chain, cannot import local components from remote components")
 		}
 
 		var pkg types.ZarfPackage
 
+		var relativeToHead string
 		if isLocal {
 			history = append(history, node.Import.Path)
-			relativeToHead := filepath.Join(history...)
+			relativeToHead = filepath.Join(history...)
 
 			// prevent circular imports (including self-imports)
 			// this is O(n^2) but the import chain should be small
 			prev := node
 			for prev != nil {
 				if prev.relativeToHead == relativeToHead {
-					return nil, fmt.Errorf("detected circular import chain: %s", strings.Join(history, " -> "))
+					return ic, fmt.Errorf("detected circular import chain: %s", strings.Join(history, " -> "))
 				}
 				prev = prev.prev
 			}
 
 			// this assumes the composed package is following the zarf layout
 			if err := utils.ReadYaml(filepath.Join(relativeToHead, layout.ZarfYAML), &pkg); err != nil {
-				return nil, err
+				return ic, err
 			}
 		} else if isRemote {
+			relativeToHead = node.Import.URL
 			remote, err := ic.getRemote(node.Import.URL)
 			if err != nil {
-				return nil, err
+				return ic, err
 			}
-			// When it's a bad file, the error gets tracked here
-			// Maybe we add something like couldn't fetch zarf yaml to this
-			// So it's clearer to the user the pull failed
 			pkg, err = remote.FetchZarfYAML()
 			if err != nil {
-				return nil, err
+				return ic, err
 			}
 		}
 
 		name := node.ImportName()
 
 		found := []types.ZarfComponent{}
-		var index int
+		index := []int{}
 		for i, component := range pkg.Components {
 			if component.Name == name && CompatibleComponent(component, arch, flavor) {
 				found = append(found, component)
-				index = i
+				index = append(index, i)
 			}
 		}
 
 		if len(found) == 0 {
-			if isLocal {
-				return nil, fmt.Errorf("component %q not found in %q", name, filepath.Join(history...))
-			} else if isRemote {
-				return nil, fmt.Errorf("component %q not found in %q", name, node.Import.URL)
-			}
+			return ic, fmt.Errorf("component %q not found in %q", name, relativeToHead)
 		} else if len(found) > 1 {
-			if isLocal {
-				return nil, fmt.Errorf("multiple components named %q found in %q satisfying %q", name, filepath.Join(history...), arch)
-			} else if isRemote {
-				return nil, fmt.Errorf("multiple components named %q found in %q satisfying %q", name, node.Import.URL, arch)
-			}
+			return ic, fmt.Errorf("multiple components named %q found in %q satisfying %q", name, relativeToHead, arch)
 		}
```

Comment:
why were these `is*` checks removed?

Reply:
I ended up setting relativeToHead on oci components to be node.Import.URL of the previous node. By doing that this block of code becomes the same in both paths. Additionally it makes the logic in lint simpler as I no longer have to backtrack on the list to get the location of the component.

Though I guess that makes relativeToHead a misnomer. Maybe `relativePathOrUrl`